example_test.go

```diff
@@ -221,52 +217,71 @@ func ExampleNetworkInterface_rateLimiting() {
 	}
 }
 
-func ExampleJailerCommandBuilder() {
+func ExampleJailerConfig_enablingJailer() {
 	ctx := context.Background()
-	// Creates a jailer command using the JailerCommandBuilder.
-	b := firecracker.NewJailerCommandBuilder().
-		WithID("my-test-id").
-		WithUID(123).
-		WithGID(100).
-		WithNumaNode(0).
-		WithExecFile("/usr/local/bin/firecracker").
-		WithChrootBaseDir("/tmp").
-		WithStdout(os.Stdout).
-		WithStderr(os.Stderr)
-
-	const socketPath = "/tmp/firecracker/my-test-id/api.socket"
+	vmmCtx, vmmCancel := context.WithCancel(ctx)
+	defer vmmCancel()
 
-	cfg := firecracker.Config{
-		SocketPath:      socketPath,
-		KernelImagePath: "./vmlinux",
-		Drives: []models.Drive{
-			models.Drive{
-				DriveID:      firecracker.String("1"),
-				IsRootDevice: firecracker.Bool(true),
-				IsReadOnly:   firecracker.Bool(false),
-				PathOnHost:   firecracker.String("/path/to/root/drive"),
-			},
-		},
+	const id = "my-jailer-test"
+	const path = "/path/to/jailer-workspace"
+	pathToWorkspace := filepath.Join(path, "firecracker", id)
+	const kernelImagePath = "/path/to/kernel-image"
+
+	uid := 123
+	gid := 100
+
+	fcCfg := firecracker.Config{
+		SocketPath:      "api.socket",
+		KernelImagePath: kernelImagePath,
+		KernelArgs:      "console=ttyS0 reboot=k panic=1 pci=off",
+		Drives:          firecracker.NewDrivesBuilder("/path/to/rootfs").Build(),
+		LogLevel:        "Debug",
 		MachineCfg: models.MachineConfiguration{
-			VcpuCount: firecracker.Int64(1),
+			VcpuCount:  firecracker.Int64(1),
+			HtEnabled:  firecracker.Bool(false),
+			MemSizeMib: firecracker.Int64(256),
+		},
+		EnableJailer: true,
+		JailerCfg: firecracker.JailerConfig{
+			UID:            &uid,
+			GID:            &gid,
+			ID:             id,
+			NumaNode:       firecracker.Int(0),
+			ChrootBaseDir:  path,
+			ChrootStrategy: firecracker.NewNaiveChrootStrategy(pathToWorkspace, kernelImagePath),
 		},
-		DisableValidation: true,
 	}
 
-	// Passes the custom jailer command into the constructor
-	m, err := firecracker.NewMachine(ctx, cfg, firecracker.WithProcessRunner(b.Build(ctx)))
+	// Check if kernel image is readable
+	f, err := os.Open(fcCfg.KernelImagePath)
 	if err != nil {
-		panic(fmt.Errorf("failed to create new machine: %v", err))
+		panic(fmt.Errorf("Failed to open kernel image: %v", err))
+	}
+	f.Close()
+
+	// Check each drive is readable and writable
+	for _, drive := range fcCfg.Drives {
+		drivePath := firecracker.StringValue(drive.PathOnHost)
+		f, err := os.OpenFile(drivePath, os.O_RDWR, 0666)
+		if err != nil {
+			panic(fmt.Errorf("Failed to drive: %v", err))
```

Comment:
`Failed to open drive for reading and writing`?

Reply:
Fixed!